docs/docs-beta/docs/guides/quality-testing/asset-checks.md

```diff
@@ -1,4 +1,63 @@
 ---
 title: "Test assets with Asset Checks"
 sidebar_position: 10
----
\ No newline at end of file
+---
+
+Asset checks in Dagster provide a way to define and execute different types of data quality checks on your data assets directly in Dagster.
+
+This guide covers the most common use cases for testing assets with asset checks and taking action based on the result.
+
+<details>
+<summary>Prerequisites</summary>
+- Familiarity with [Assets](/concepts/assets)
+</details>
+
+## Testing assets with a single asset check
+
+The example below defines a single asset check on an asset that fails if the `order_id` column of the asset contains a null value.
+
+In this example, the asset check will run after the asset has been materialized, to ensure the quality of its data.
+
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/single-asset-check.py" language="python" title="Asset with a single asset check" />
+
+## Testing assets with multiple asset checks
+
+In most cases, checking the data quality of an asset will require multiple checks.
+
+The example below defines two asset checks using the `multi_asset_check` decorator:
+- one that fails if the `order_id` column of the asset contains a null value.
+- another one that fails if the `item_id` column of the asset contains a null value.
+
+In this example, both asset check will run in a single operation after the asset has been materialized.
+
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/multiple-asset-checks.py" language="python" title="Asset with multiple asset checks" />
+
+## Defining asset checks using a factory pattern
+
+Defining multiple checks can also be done using a factory pattern. The example below defines the same two asset checks as in the previous example, but this time using a factory pattern and the `@asset_check` decorator.
```

Comment:
Curious why you chose this over a `multi_asset_check`. Wouldn't this run `n` _ops_, one for each check?

Reply:
That's a good call, I've updated the example to use multi_asset_check [here](https://github.com/dagster-io/dagster/pull/24035/commits/5c38554cca8501735ad628bb6157a7cd58bbb583)